nupic/research/frameworks/pytorch/imagenet_experiment.py

```diff
@@ -556,16 +570,20 @@ def validate(self, loader=None):
         return results
 
     def train_epoch(self, epoch):
-        train_model(
-            model=self.model,
-            loader=self.train_loader,
-            optimizer=self.optimizer,
-            device=self.device,
-            criterion=self.loss_function,
-            batches_in_epoch=self.batches_in_epoch,
-            pre_batch_callback=functools.partial(self.pre_batch, epoch=epoch),
-            post_batch_callback=functools.partial(self.post_batch, epoch=epoch),
-        )
+        with torch.autograd.profiler.profile(use_cuda=torch.cuda.is_available(),
```

Comment:
Hopefully this won't add extra overhead when profile==False?

Reply:
When `enabled=False` this context just returns.
See https://pytorch.org/docs/stable/autograd.html#torch.autograd.profiler.profile